### PR TITLE
Cherry-pick - GFX942 Fix & Ubuntu 24.04 Failures Fix (#1450)

### PR DESCRIPTION
**Pull request by kiritigowda**
* boundary check for canny
* AMD Clang++ - Default compiler

### CODE REVIEW

**Review by hansely — DISMISSED** [review]

**Review by LakshmiKumar23 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,43 +5,40 @@
 Documentation for MIVisionX is available at
 [https://rocm.docs.amd.com/projects/MIVisionX/en/latest/doxygen/html/index.html](https://rocm.docs.amd.com/projects/MIVisionX/en/latest/doxygen/html/index.html)
 
-## MIVisionX 3.1.0 (unreleased)
+## (unreleased) MIVisionX 3.2.0
```
Comment (spolifroni-amd) [inline CHANGELOG.md:8]:
```suggestion
## (Unreleased) MIVisionX 3.2.0
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,43 +5,40 @@
...
+* Hardware decode requires the ROCm `graphics` usecase.
+
+### Upcoming changes
+
+* Optimized audio augmentations support for VX_RPP
+
+
+## MIVisionX 3.1.0 for ROCm 6.3.1
 
 ### Changes
```
Comment (spolifroni-amd) [inline CHANGELOG.md:26]:
```suggestion
### Changed
```

**Review by spolifroni-amd — CHANGES_REQUESTED** [review]
Small changes in the changelog but they need to be done for the release notes script.

**Comment by kiritigowda** [comment]
> Small changes in the changelog but they need to be done for the release notes script.

@spolifroni-amd changes accepted

**Review by spolifroni-amd — APPROVED** [review]

**Review by hansely — APPROVED** [review]

**Review by LakshmiKumar23 — APPROVED** [review]

**Review by rrawther — APPROVED** [review]

**Pull request merged**